ta: Add Android.mk

vchong opened the pull request:
Signed-off-by: Victor Chong <victor.chong@linaro.org>

Commit:
ta: Add support for Android.mk

To allow hello_world to build properly in aosp.

Signed-off-by: Victor Chong <victor.chong@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>

jockebech commented on ta/Makefile:
```diff
@@ -4,25 +4,4 @@ CPPFLAGS += -DCFG_TEE_TA_LOG_LEVEL=$(CFG_TEE_TA_LOG_LEVEL)
 # The UUID for the Trusted Application
 BINARY=8aaaf200-2450-11e4-abe2-0002a5d5c51b
 
--include $(TA_DEV_KIT_DIR)/mk/ta_dev_kit.mk
```
Why are you removing non-Android stuff in an "Add Android.mk" patch? Follow up, I remember the discussion about the targets below. How come you want to removed them now?

jforissier replied:
@jbech-linaro `all:` and `clean:` are defined in `ta_dev_kit.mk`. But I agree, removing this should be a separate commit.

vchong replied:
Ok. I'll move this to a separate commit.

The `all` and `clean` here overwrite those in `ta_dev_kit.mk`, causing a `No rule to make target '8aaaf200-2450-11e4-abe2-0002a5d5c51b.ta', needed by 'all'. ` error in the android build.

In trying to understand the error, it seems that `ta/Android.mk` calls `$(BUILD_OPTEE_MK)` which is `optee_os/mk/aosp_optee.mk` which calls `ta/Makefile` which calls `ta_dev_kit.mk` and friends and it seems like `8aaaf200-2450-11e4-abe2-0002a5d5c51b.ta` is not a valid target for `ta_dev_kit.mk` and friends.

If I change `all: $(BINARY).ta` to `all: $(link-out-dir)/$(BINARY).ta` here then there's no error in the android build since it's now a valid target for `ta_dev_kit.mk` and friends, namely `link.mk`.

BUT, why does `$(BINARY).ta` not cause a `No rule..` error when we use `build.git`? Any ideas?

vchong replied:
@jforissier @jbech-linaro Commit split into two. Ok to move forward?

vchong replied:
`$(BINARY).ta` DOES cause a `No rule..` error in `build.git` if `O` is specified and not set to `.`, which is the case for the Android build.

vchong replied:
From https://github.com/OP-TEE/optee_os/pull/1276 (when a target starts with `./`, `$@` does not), it seems like above is expected behavior?

When `O` is not specified (or specified as `.`), `out-dir` is `.`, and in `<ta_dev_kit_dir>/mk/link.mk`, `link-out-dir = $(out-dir)` and the `$(link-out-dir)/$(binary).*` targets just become `./$(binary).*`, so `$(BINARY).ta` = `$(binary).ta` = `./$(binary).ta` is a valid target?

But when `O` is specified as something other than `.`, the there's no such target in `link.mk`.

Commit:
ta:Makefile: Remove 'all' and 'clean' targets

'all:' and 'clean:' are defined in ta_dev_kit.mk, so don't need them
here since they will just overwrite the ones in ta_dev_kit.mk and
possibly cause issues. The Makefiles in optee_test/ta/*/ don't have
them as well so keep things consistent.

Signed-off-by: Victor Chong <victor.chong@linaro.org>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>

jockebech commented:
Fine with me

jforissier commented:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

vchong commented:
Tags added. Please merge if ok. Thanks!
